File: app/src/main/java/com/wegielek/signalychinese/models/ChineseTextInputEditText.java
package com.wegielek.signalychinese.models;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatEditText;

import com.wegielek.signalychinese.interfaces.SearchTextBoxListener;


public class ChineseTextInputEditText extends AppCompatEditText {

    private SearchTextBoxListener mListener;

    public ChineseTextInputEditText(@NonNull Context context) {
        super(context);
    }

    public ChineseTextInputEditText(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        //setBackgroundTintList(context.getColorStateList(R.color.selector_edit_text));
    }

    public ChineseTextInputEditText(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    public void setOnSelectionChangedListener(SearchTextBoxListener searchTextBoxListener) {
        mListener = searchTextBoxListener;
    }

    @Override
    protected void onSelectionChanged(int selStart, int selEnd) {

        if(mListener != null) {
            mListener.onSelectionChanged(selStart, selEnd);
        }

        super.onSelectionChanged(selStart, selEnd);
    }



}

File: app/src/main/java/com/wegielek/signalychinese/classes/ChineseTextInputEditText.kt
package com.wegielek.signalychinese.classes

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatEditText
import com.wegielek.signalychinese.interfaces.SearchTextBoxListener

class ChineseTextInputEditText : AppCompatEditText {
    private var mListener: SearchTextBoxListener? = null

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    fun setOnSelectionChangedListener(searchTextBoxListener: SearchTextBoxListener) {
        mListener = searchTextBoxListener
    }

    override fun onSelectionChanged(selStart: Int, selEnd: Int) {
        mListener?.onSelectionChanged(selStart, selEnd)
        super.onSelectionChanged(selStart, selEnd)
    }
}